pex/pex.py

```diff
@@ -617,10 +617,47 @@ def execute_script(self, script_name):
         dist_script = get_script_from_distributions(script_name, dists)
         if not dist_script:
             return "Could not find script {!r} in pex!".format(script_name)
-        TRACER.log("Found script {!r} in {!r}.".format(script_name, dist))
-        return self.execute_content(
-            dist_script.path, dist_script.read_contents(), argv0=script_name
-        )
+        TRACER.log("Found script {!r} in {!r}.".format(script_name, dist_script.dist))
+
+        contents = dist_script.read_contents()
+        shebang = contents.split("\n", 1)[0]
+        try:
+            interpreter = PythonInterpreter.from_shebang(shebang)
+            if interpreter and interpreter.version_string != self.interpreter.version_string:
+                TRACER.log(
+                    "Python script shebang resolved to {shebang_python}, but we will execute it with {pex_python}".format(
+                        shebang_python=interpreter.version_string,
+                        pex_python=self.interpreter.version_string,
+                    ),
+                    V=2,
+                )
+        except PythonInterpreter.InterpreterNotFound as e:
+            TRACER.log(
+                "Python script shebang indicated unknown interpreter: {shebang!r}: {e}, will execute it with {version}.".format(
+                    shebang=shebang, e=e, version=self.interpreter.version_string
+                ),
+                V=2,
+            )
+            interpreter = self.interpreter
+
+        if interpreter:
+            TRACER.log(
+                "Executing python script with {}".format(self.interpreter.version_string), V=2
+            )
+            return self.execute_content(dist_script.path, contents, argv0=script_name)
+        else:
+            TRACER.log("Executing non-python script using shebang: {}".format(shebang), V=2)
+            return self.execute_external(dist_script.path)
+
+    def execute_external(self, cmd):
+        # type: (str) -> Any
+        try:
+            # execute as a shell command, inheriting stdout and stderr
+            proc = Executor.open_process([cmd] + sys.argv[:1], shell=True)
+            proc.communicate()
```

Comment:
You can just `return proc.wait()` instead of `proc.communicate(); return proc.returncode` - you do nothing with the return of `proc.communicate()` which is a hint.